File: Solutions/GuidedSolutions/SayHello/SayHello.c
#include <stdio.h>

void sayHello(char* name) {
	//print hello, name
	printf("Hello, %s", name);
}

int main() {
	//create variable to read name into
	char name[35];

	//prompt user for name
	printf("Enter your name(under 35 characters): ");
	
	//read name from input
	if(fgets(name, sizeof(name), stdin))	
		sayHello(name);
	else
		return 1; //return with error if name was not read correctly

	//return without error
	return 0;
}

File: Solutions/GuidedSolutions/SayHello/SayHello.java
import java.util.Scanner;

public class SayHello {
	public static void sayHello(String name) {
		//print passed name
		System.out.println("Hello, " + name + "!");
	}

	
	public static void main(String[] args) {
		//scanner to read user input
		Scanner in = new Scanner(System.in);

		//prompt user for their name and read their input
		System.out.println("Enter your name: ");
		String name = in.nextLine();

		//call sayHello with user's input
		sayHello(name);
	}
}
